src/AdoNetCore.AseClient/AseConnection.cs

```diff
@@ -539,7 +539,37 @@ public static void ClearPools()
 
         public bool IsCaseSensitive()
         {
-            return _internal?.IsCaseSensitive() ?? false;
+            int result;
+            using (var command = CreateCommand())
+            {
+                command.CommandType = CommandType.Text;
+                command.CommandText = "SELECT value FROM master.dbo.sysconfigures WHERE name = 'default sortorder id'";
+
+                result = (int)command.ExecuteScalar();
+            }
+
+            switch (result)
```

Comment:
`HashSet<int>.Contains(...)` might be more appropriate than a switch

Reply:
Disagree. If you're motivated by performance:

1. I would hazard that the raw performance of a small switch is faster than looking up within a HashSet. To get access to that constant lookup speed you have to jump down at least one stack frame. The cost of instantiating the HashSet would certainly be more, so that would have to be done statically.
2. The instantiation cost and memory overhead of a static HashSet would be all the time. And this operation is likely to never be called at all.

Reply:
OK. Did the science. 10m iterations of each is pretty consistently 390ms for hashset, and 250ms for switch. Not the best test harness. Also tested binary search of an ordered array which was worst of all at 540ms.

Reply:
👍